Emergence/Contexts/ArtworkViewController.swift

```diff
@@ -21,8 +22,30 @@ class ArtworkViewController: UIViewController {
         artworkDimensionsInchesLabel.text = artwork.dimensionsInches
         artworkDimensionsCMsLabel.text = artwork.dimensionsCM
 
+        /// OK, we want to re-use images as much as possible
+        /// so that you're not staring at grey boxes.
+
+        /// In this case we are grabbing the thumbnail, probably loaded
+        /// in the ShowViewController and seeing if it's in the image cache, if it 
+        /// is then we set the placeholder to be the existing image
+        /// and then request a bigger image async.
+
+        let showArtworksHeight:CGFloat = 620
+
         if let defaultImage = artwork.defaultImage, let actualImage = defaultImage as? Image {
-            artworkPreviewImage.ar_setImage(actualImage, height: artworkPreviewImage.bounds.height)
+
+            let height = artworkPreviewImage.bounds.height
+            guard let fullThumbnailPath = actualImage.bestThumbnailWithHeight(height) else { return print("no thumbnail for artwork") }
+
+            let manager = SDWebImageManager.sharedManager()
+            if let previousThumbnail = actualImage.bestThumbnailWithHeight(showArtworksHeight) where manager.cachedImageExistsForURL(previousThumbnail) {
+                let key = manager.cacheKeyForURL(previousThumbnail)
+                let smallerInitialThumbnail = manager.imageCache.imageFromMemoryCacheForKey(key)
+
+                artworkPreviewImage.sd_setImageWithURL(fullThumbnailPath, placeholderImage: smallerInitialThumbnail)
```

Comment:
Too bad SDWebImage doesn't have this built-in. Something like "load this image URL, but if you don't have it cached you can use this other image URL in the mean time."